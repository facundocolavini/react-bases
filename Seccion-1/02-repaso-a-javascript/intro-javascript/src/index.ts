'estoy en async await seccion 3'
export {}