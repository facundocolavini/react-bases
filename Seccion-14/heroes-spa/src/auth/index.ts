export * from './pages';
export * from './routes/AuthRoutes';
export * from './context';