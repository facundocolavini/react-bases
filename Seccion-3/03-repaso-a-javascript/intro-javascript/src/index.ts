
console.log('Fin de la seccion 3');

export {}