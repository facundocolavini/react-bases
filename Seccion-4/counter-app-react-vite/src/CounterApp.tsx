import PropTypes from 'prop-types'
import { useState } from 'react';

type ValuesProps  = {
   value: number;

}

export const CounterApp = ({value}: ValuesProps): JSX.Element => {
    
    const [counter , setCounter] = useState(value)

    const handleAdd =(): void => {
        // 
        setCounter(counter + 1)
        //Otra forma :
        // setCounter((c)=>  c + 1) // valor actual del counter + 1
    }

    const handleSubstract = (counter: number): void => {
        setCounter(counter - 1)
    }

    const handleReset = (): void=>{
        setCounter(value)
    }
    
    return (
        <>
            <h1>CounterApp</h1>
            <h2 data-testid="counter">{counter}</h2>
            <button aria-label="btn-increment" onClick={handleAdd}>+1</button>
            <button aria-label="btn-decrement" onClick={()=>handleSubstract(counter)}>-1</button>
            <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
        </>
    )
}

CounterApp.propTypes ={
    value: PropTypes.number.isRequired,
}