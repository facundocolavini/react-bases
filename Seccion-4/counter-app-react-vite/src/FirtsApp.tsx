//Proptypes 
import PropTypes from 'prop-types'


// Variables

const newMessage : string ='Facundo';// No se rerenderiza
const newArray : Array<number> = [1,2,3,4,5,6,7,8,9];// No se rerenderiza
interface Message {
    message: string;
    title: string;
}
const msgObj: Message = {
    message: 'Hola Mundo',
    title: 'Facundo'
}
//Recomendado si no tiene una dependencia del componente dejar las funciones por fuera 
function sayHi(message:string): string{
        return message 
}


// Doble validacion si es que hay propTypes por eso hago las props opcionales ?
interface Titles {
    title?: string;
    subTitle?: string; 
    name?: string;
}

export const FirtsApp: React.FC<Titles> = ({title,subTitle, name}) => {
/*     if(!title) {
        throw new Error('El titulo no existe');
    } */
    
    return (
    <>
        <h1>{title}</h1>
        {/* <code>{ JSON.stringify(msgObj) }</code> */}
        <p>{subTitle}</p>
        <p>{name}</p>
    </>
  )
}

// PropTypes
FirtsApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired, // Si uso interfaces y propTypes estaria manejando 2 veces la validacion de props
}

//Default Propts entran antes de nuestro propTypes
FirtsApp.defaultProps = { // Si utilizo interface al poner valores default que no existen en mi interface dara un error
    title: 'No hay titulo',
    subTitle: 'No hay subtitulo',
    name: 'Facundo Colavini'
}

