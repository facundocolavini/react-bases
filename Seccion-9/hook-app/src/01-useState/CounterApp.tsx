import { SetStateAction, useState, Dispatch } from 'react';
import * as React from 'react';

type Props = {}

type CounterStateType = {
    counter1: number;
    counter2: number;
    counter3: number;
}
export default function CounterApp({}: Props) {
//   const [counter, setCounter] = useState(10)
  // Inicializando el estado con un objeto 
  const [{counter1, counter2, counter3}, setCounters] = useState({
        
        counter1:10,
        counter2:20,
        counter3:30,
    })


    const onAddCounters = () =>{
        setCounters(({counter1, counter2, counter3}) => ({
                counter1:counter1 + 1,
                counter2,
                counter3
            }
        ));
    }

    return (
    <>
        <h1>Counter 1: { counter1 }</h1>
        <h1>Counter 2: { counter2 }</h1>
        <h1>Counter 3: { counter3 }</h1>
        <hr />
        <button className="btn btn-primary" onClick={onAddCounters}>+1</button>
    </>
  )
}