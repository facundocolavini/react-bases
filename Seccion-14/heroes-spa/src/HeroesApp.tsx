import { AuthProvider } from "./auth"
import { AppRouter } from "./router/AppRouter"

interface AppHeroes {

}

export const HeroesApp = (props: AppHeroes) => {
  return (
    <AuthProvider>
      <AppRouter />
    </AuthProvider>
  )
}