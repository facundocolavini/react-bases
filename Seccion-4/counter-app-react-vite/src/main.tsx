import React from 'react';
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp';
import { FirtsApp } from './FirtsApp';
import { HelloWorldApp } from './HelloWorldApp';
//Styles 
import './styles.css'




ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
    <React.StrictMode>
        {/* <FirtsApp></FirtsApp> */}
        <CounterApp value={1} />
    </React.StrictMode>
  )