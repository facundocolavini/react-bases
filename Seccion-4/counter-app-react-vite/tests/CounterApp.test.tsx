
import * as React from 'react'
import { describe, expect, test } from '@jest/globals';
import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';



describe('Trabajando en el componente CounterApp', () => {
    const initialValue: number = 10; 
    const newReset: number = 100;

    test("Debe de hacer match con el snapshot", () => {
        const { container } =  render(<CounterApp value={initialValue}/>)
        expect(container).toMatchSnapshot()
    }) 

    test("El contador debe de mostrar como valor inicial 100 en su elemento H2", ()=>{
        render(<CounterApp value={100}></CounterApp>)

        expect(screen.getByText(newReset)).toBeTruthy()
        expect(screen.getByRole('heading',{level:2}).innerHTML).toContain('100')
    })

    // Testing de botones e interacciones
    test("Debe de incrementar con el boton +1",()=>{
        // Sujeto de pruebas
        render(<CounterApp value={initialValue}></CounterApp>)

        //Obtenemos
        const btnIncrement: HTMLElement = screen.getByText('+1')
        
        //Evento  Click (Simulo darle click al boton incrementar)
        fireEvent.click( btnIncrement )
        
        //Ascercion 
        expect(screen.getByText('11')).toBeTruthy()
        screen.debug()
    })

    test("Debe de decrementar con el boton -1",()=>{
        // Sujeto de pruebas
        render(<CounterApp value={initialValue}></CounterApp>)

        //Obtenemos
        const btnDecrement: HTMLElement = screen.getByText('-1')
        
        //Evento  Click (Simulo darle click al boton incrementar)
        fireEvent.click( btnDecrement )
        
        //Ascercion 
        expect(screen.getByText('9')).toBeTruthy()

        screen.debug()
    })

    test("Debe de reiniciar el contador al clickear el boton reset",()=>{
        render(<CounterApp value={initialValue}></CounterApp>)
        
        // const btnIncrement: HTMLElement = screen.getByText('+1') //Sin arial-label
        // const btnResetCounter: HTMLElement = screen.getByText('Reset')
        const btnIncrement: HTMLElement = screen.getByRole('button', { name: 'btn-increment' })
        const btnResetCounter: HTMLElement = screen.getByRole('button', { name: 'btn-reset' })
        const counter: HTMLHeadElement = screen.getByTestId('counter')// Con data-testid
        
        fireEvent.click( btnIncrement ) 
        fireEvent.click( btnIncrement ) 
        fireEvent.click( btnIncrement ) 
        
        // Reset Button
        fireEvent.click(btnResetCounter)
        // expect(screen.getByRole('heading', { level: 2}).innerHTML).toContain(initialValue.toString())
        expect(counter.innerHTML).toContain(initialValue.toString())
       
        screen.debug()
    })
})



