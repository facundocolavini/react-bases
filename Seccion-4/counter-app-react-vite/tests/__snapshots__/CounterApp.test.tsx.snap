// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trabajando en el componente CounterApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    data-testid="counter"
  >
    10
  </h2>
  <button
    aria-label="btn-increment"
  >
    +1
  </button>
  <button
    aria-label="btn-decrement"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
